source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,377 @@
+        $syncSchedule = $apProps.PeriodicCatalogUpdateSchedule
+        $status       = 'Present'
+
+        $convertCimParam = @{
+            ScheduleString = $syncSchedule
+            CimClassName   = 'DSC_CMAssetIntelligenceSynchronizationSchedule'
+        }
+        $sSchedule = ConvertTo-CimCMScheduleString @convertCimParam
+    }
+
+    else
+    {
+        $status = 'Absent'
+    }
+
+    return @{
+        SiteServerName        = $SiteServerName
+        SiteCode              = $SiteCode
+        CertificateFile       = $cert
+        Enable                = $apEnabled
+        EnableSynchronization = $syncEnabled
+        Schedule              = $sSchedule
+        Ensure                = $status
+    }
+}
+
+ <#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .PARAMETER CertificateFile
+        Specifies the path to a System Center Online authentication certificate (.pfx) file.
+        If used, this must be in UNC format. Local paths are not allowed.
+
+    .PARAMETER Schedule
+        Specifies when the asset intelligence catalog is synchronized.
+
+    .PARAMETER Enable
+        Specifies whether the installed asset intelligence role is enabled or disabled.
+
+    .PARAMETER EnableSynchronization
+        Specifies whether to synchronize the asset intelligence catalog.
+
+    .PARAMETER Ensure
+        Specifies whether the asset intelligence synchronization point is present or absent.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [String]
+        $CertificateFile,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $Schedule,
+
+        [Parameter()]
+        [Boolean]
+        $Enable,
+
+        [Parameter()]
+        [Boolean]
+        $EnableSynchronization,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+
+    try
+    {
+        if ($Ensure -eq 'Present')
+        {
+            if ($state.Ensure -eq 'Absent')
+            {
+                Write-Verbose -Message ($script:localizedData.AddAPRole -f $SiteServerName)
+                Add-CMAssetIntelligenceSynchronizationPoint -SiteSystemServerName $SiteServerName
```

Comment:
if the server isn't a site system server this command does not work.  Brand new server and command fails. once I ran
New-CMSiteSystemServer -SiteCode $SiteCode -SiteSystemServerName $SiteServerName
I was able to run the command above on line 151

Reply:
Added logic for adding if server isn't already a site system server. updated strings file to match and added to pester to match.